### PR TITLE
chore: add build id to settings page

### LINKED ISSUE #3905: [CW-370] Add a build ID to distinguish deployments based on develop branch
**Is your feature request related to a problem? Please describe.**
Currently, continuous deployments using the `develop` branch are not distinguishable since we don't create a tag for every merge to develop. 

**Describe the solution you'd like**
Generate a build number that is incremented for every merge to develop/master. Have this be displayed next to the version on the settings page. 

**Describe alternatives you've considered**
Use the commit hash as the build ID.

**Additional context**

image

<sub>[CW-370](https://linear.app/chatwoot/issue/CW-370/add-a-build-id-to-distinguish-deployments-based-on-develop-branch)</sub>

### PR DESCRIPTION
**Pull request by vishnu-narayanan**
# Pull Request Template

## Description

chore: add build Id to settings page

image

Fixes #3905

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -7,5 +7,8 @@ npx --no-install lint-staged
 # lint only staged ruby files
 git diff --name-only --cached | xargs ls -1 2>/dev/null | grep '\.rb$' | xargs bundle exec rubocop --force-exclusion -a
 
+# echo git-sha to file
```
Comment (pranavrajs) [inline .husky/pre-commit:10]: @vishnu-narayanan Can we make this get this dynamically in the initializer? By making it a commit hook, we would need to ensure that the git-sha is updated every time a PR is raised.





Can we do something like this? https://stackoverflow.com/a/33380485

```
running_rake = ::File.basename($0) == 'rake'

unless running_rake
  latest_commit = `git rev-parse HEAD`
  STDOUT.puts "**** latest commit: #{latest_commit}"
end
```

Comment (vishnu-narayanan) [reply .husky/pre-commit]: @pranavrajs Gotcha. Will make the changes.

**Comment by vishnu-narayanan** [comment]
@pranavrajs I have removed the pre-commit hook approach.

Modifying it to be

1. Directly parse SHA if git rev-parse is available.
2. Fallback to the `.git_sha` file if 1 fails.
3. Return unknown if 1 and 2 fail.

Option 1 with `git rev-parse HEAD` would only work if the `.git` directory is available. We are stripping this off in docker, k8s, and Heroku deployments. We could modify the Procfile to generate `.git_sha` for Heroku-based deployment and the `Dockerfile` to generate `.git_sha` for Docker-based deployments.  

TLDR; Option 1 for Linux deployments while falling back to option 2 for other implementations.

Thoughts?

cc: @sojan-official

**Review by sojan-official — COMMENTED** [review]
The approach suggested sounds good to me

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Please see the comment above.

File: .git-sha [hunk 2]
```diff
@@ -0,0 +1 @@
+dd40d96776d1fcfa326f30edb27dca2af141077e
```
[the remark this answers is not in the document]
Comment (vishnu-narayanan) [reply .git-sha]: Yes, deleted it.

File: app/controllers/dashboard_controller.rb [hunk 3]
```diff
@@ -56,7 +56,19 @@ def app_config
       FB_APP_ID: GlobalConfigService.load('FB_APP_ID', ''),
       FACEBOOK_API_VERSION: 'v14.0',
       IS_ENTERPRISE: ChatwootApp.enterprise?,
-      AZURE_APP_ID: ENV.fetch('AZURE_APP_ID', '')
+      AZURE_APP_ID: ENV.fetch('AZURE_APP_ID', ''),
+      GIT_SHA: fetch_git_sha
     }
   end
+
+  def fetch_git_sha
```
[the remark this answers is not in the document]
Comment (vishnu-narayanan) [reply app/controllers/dashboard_controller.rb]: Done
